src/OutputReaders/field_time_series.jl

```diff
@@ -180,6 +182,34 @@ function Base.getindex(fts::FieldTimeSeries{LX, LY, LZ, OnDisk}, n::Int) where {
     return Field(loc, fts.grid; indices=fts.indices, boundary_conditions=fts.boundary_conditions, data=field_data)
 end
 
+# Linear time interpolation
+function Base.getindex(fts::FieldTimeSeries, time_index::Time)
+    Ntimes = length(fts.times)
+    time = time_index.time
+    n₁, n₂ = index_binary_search(fts.times, time, Ntimes)
+    if n₁ == n₂ # no interpolation
+        return fts[n₁]
+    end
+    
+    # fractional index
+    @inbounds n = (n₂ - n₁) / (fts.times[n₂] - fts.times[n₁]) * (time - fts.times[n₁]) + n₁
+    return compute!(Field(fts[n₂] * (n - n₁) + fts[n₁] * (n₂ - n)))
+end
+
+# Linear time interpolation
+function Base.getindex(fts::FieldTimeSeries, i::Int, j::Int, k::Int, time_index::Time)
+    Ntimes = length(fts.times)
+    time = time_index.time
+    n₁, n₂ = index_binary_search(fts.times, time, Ntimes)
+    if n₁ == n₂ # no interpolation
+        return getindex(fts, i, j, k, n₁)
+    end
+    
+    # fractional index
+    @inbounds n = (n₂ - n₁) / (fts.times[n₂] - fts.times[n₁]) * (time - fts.times[n₁]) + n₁
+    return getindex(fts, i, j, k, n₂) * (n - n₁) + getindex(fts, i, j, k, n₁) * (n₂ - n)
+end
```

Comment:
```suggestion
   
    # fractional index
    @inbounds n = (n₂ - n₁) / (fts.times[n₂] - fts.times[n₁]) * (time - fts.times[n₁]) + n₁
   fts_interpolated = getindex(fts, i, j, k, n₂) * (n - n₁) + getindex(fts, i, j, k, n₁) * (n₂ - n)
   
   # Don't interpolate if n = 0.
   return ifelse(n₁ == n₂, getindex(fts, i, j, k, n₁), fts_interpolated)
end
```

Reply:
I prefer not to do the interpolation if it is not required

Reply:
Why?

Reply:
Ah, I was actually looking at the other getindex, no in the kernel it's ok

Reply:
(It's not good practice to resolve other people's comments, unless they are not responding or something --- the reviewer gets priority to decide whether their comment has been resolved or not.)

Reply:
The main concern here is performance for time interpolation of `FieldTimeSeries` within a `Forcing`, which enters into the expensive 3D kernels to calculate tendencies. This will be an important case; for example for driving LES using 3D forcing derived from GCM. We can't know the flip side either way without profiling or benchmarking, but `ifelse` is simply a conservative choice to avoid performance cliffs (or at least, that has been our experience time and time again in the past --- maybe there is new information we should consider?)

Reply:
yeah, the kernel is fast so it shouldn't be a problem. On the other hand, a simple if condition should be okay here as there should not be problems of branch divergence (all threads should look at the same time). Still, I m okay with interpolating in the kernel

Reply:
What kernel is fast? I'm referring to the situation where this is embedded in another kernel. For example, this `getindex` call can be embedded in the kernels that calculate model tendencies.

Reply:
> On the other hand, a simple if condition should be okay here as there should not be problems of branch divergence (all threads should look at the same time).

Got it. Is that still the case in very complicated kernels, for example kernels that calculate 3D tendencies?

Reply:
sorry, linear interpolation in GPU kernels is fast. That is if threads access the same time index. There could be a situation in which different threads access different time indexes but it doesn't immediately come to mind.

Reply:
My question is whether that principle is still useful when the linear interpolation is surrounded by lots of other expensive operations, as can happen when this is used inside a tendency kernel.

Reply:
Here's another example: `FieldTimeSeries` is used to store data for an advecting velocity field. This is the application you'd get if you're interested in, for example, advecting biogeochemical tracers with a velocity field derived from the ECCO state estimate. In that case the linear interpolation is inserted into an advection scheme.